subsys/net/lib/http_server/CMakeLists.txt

```diff
@@ -0,0 +1,29 @@
+# SPDX-License-Identifier: Apache-2.0
```

Comment:
This is more of a sample than a library at the moment. 

Having a sample is fine - and we should have one, but `subsys/` is not where it belongs. And the sample should make actual use of the library we're working on.

Generally, we really need to think on how should the HTTP2 server library look like. For sure we'll need a public API (as I wrote on Discord, the public header should be defined in `include/zephyr/net/http`. You need to think on what public functions should be exposed to the application. The most basic ones that come up to me  would be:
* `http2_server_init()`
* `http2_server_start()` 
* `http2_server_stop()`

For the resource (webpage) representation in the server, you should have a look at the API added some time ago by @cfriedt (which could likely provide more details about it): https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/net/http/service.h#L19

Finally, the sample app should make use of the above to create a HTTP2 server. And we should cover those functionalities in tests, just as listed in the GSoC umbrella issue.

Reply:
Actually, to expand on this a bit, the http service description was specifically designed to be more-or-less independent of any actual implementation details. 

What that allows us to do, for example, is to easily (for various definitions of easy) compare two or more different http server implementations in various sample applications and testsuites. There may be an implementation-specific / conditionally compiled source file here or there, but otherwise it should work.

Ultimately, this will allow us to quantitatively compare HTTP server implementations in Zephyr and provide graphs, evidence, etc to show that e.g. Emna's server implementation outperforms CivetWeb by a factor of 2 or something along those lines, or this implementation or that implementation is able to handle N concurrent clients while using M bytes of memory, and so on.

Reply:
I'll see if I can pull the CivetWeb implementation in for reference purposes, in a way that we can easily choose this or that based on Kconfig.